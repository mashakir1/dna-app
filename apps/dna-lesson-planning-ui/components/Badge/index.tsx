// Styles
import styles from "./badge.module.scss";

export const Badge: ({
  text,
  classNames,
}: {
  text: string;
  classNames?: string;
}) => JSX.Element = ({ text, classNames }) => (
  <div className={`${styles.badge} ${classNames} button`}>
    <span>{text}</span>
  </div>
);
