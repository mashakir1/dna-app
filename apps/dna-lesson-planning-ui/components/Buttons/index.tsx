export { IconLinkButton } from "./IconLinkButton";
export { OutlineLinkButton } from "./OutlineLinkButton";
export { FilledButton } from "./FilledButton";
export { FilledLinkButton } from "./FilledLinkButton";
export { EmptyButton } from "./EmptyButton";
export { EmptyIconButton } from "./EmptyIconButton";
