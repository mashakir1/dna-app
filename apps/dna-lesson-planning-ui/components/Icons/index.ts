export { ClickableIcon } from "./ClickableIcon";
export { IconWithBackground } from "./IconWithBackground";
