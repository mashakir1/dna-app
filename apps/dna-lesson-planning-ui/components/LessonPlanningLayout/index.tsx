// Imports
import { ReactNode } from "react";

// Components
import { Header } from "./Header";

// Styles
import styles from "./LessonPlanningLayout.module.scss";

const Layout = ({ children }: { children: ReactNode }) => (
  <div className={styles.page}>
    <Header />
    <div className={`${styles.container} flex column gap-l`}>{children}</div>
  </div>
);
export default Layout;
