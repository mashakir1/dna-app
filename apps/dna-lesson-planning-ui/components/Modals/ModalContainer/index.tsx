// Imports
import { ReactNode } from "react";

// Styling
import styles from "./modalContainer.module.scss"

interface ModalContainerProps {
  children: ReactNode;
  visible: boolean;
}

const ModalContainer: (props: ModalContainerProps) => JSX.Element = ({ children, visible }) => (
  <div className={`${styles.modalContainer} ${visible ? styles.visible : ""}`}>
    {children}
  </div>
);

export default ModalContainer;
