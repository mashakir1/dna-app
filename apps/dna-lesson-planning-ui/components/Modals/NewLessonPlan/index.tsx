// Imports
import { Dispatch, SetStateAction, useState } from "react";

// Components
import { Modal } from "..";
import { NewLessonPlanFirstPage } from "./firstPage";
import { NewLessonPlanSecondPage } from "./secondPage";

interface NewLessonPlanProps {
  visible: boolean;
  setVisible: Dispatch<SetStateAction<boolean>>;
}

const NewLessonPlan: (props: NewLessonPlanProps) => JSX.Element = ({
  visible,
  setVisible
}) => {
  const [pageNumber, setPageNumber] = useState<number>(0);

  return <Modal
    visible={visible}
    closeButton
    setVisible={setVisible}
    paginationProps={{ pageNumber, setPageNumber }}
  >
    {pageNumber === 0
      ? <NewLessonPlanFirstPage setPageNumber={setPageNumber} />
      : <NewLessonPlanSecondPage />}
  </Modal>;
};

export default NewLessonPlan;
