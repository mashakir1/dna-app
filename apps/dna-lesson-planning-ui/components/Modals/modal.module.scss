//TODO check % sizes
.modal {
  min-height: 400px;
  min-width: 620px;
  background-color: $colour_white;
  position: fixed;
  visibility: hidden;
  opacity: 0;
  border-radius: 16px;
  display: flex;
  padding: $spacing-l;
}

.modalContent {
  display: flex;
  align-items: center;
  width: 100%;
}

.modal .modalButton {
  background-color: $colour_white;
  border: none;

  path {
    fill: $colour_dark_text;
  }
}

.modal .closeButton {
  position: absolute;
  right: $spacing-l;
  top: $spacing-l;
}

.modal .backButton {
  position: absolute;
  left: $spacing-l;
  top: $spacing-l;
}

.visible {
  visibility: visible;
  opacity: 1;
}
