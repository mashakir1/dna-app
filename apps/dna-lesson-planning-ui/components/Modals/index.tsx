// Imports
import { Dispatch, ReactNode, SetStateAction } from "react";

// Components
import { CloseModalIcon } from "components/Svg/CloseModalIcon";
import { BackArrow } from "components/Svg/BackArrow";
import ModalContainer from "./ModalContainer";

// Styling
import styles from "./modal.module.scss";

interface PaginationProps {
  pageNumber: number;
  setPageNumber: Dispatch<SetStateAction<number>>;
}

export interface ModalProps {
  visible: boolean;
  setVisible?: Dispatch<SetStateAction<boolean>>;
  children: ReactNode;
  closeButton?: boolean;
  paginationProps?: PaginationProps
}

export const Modal: (props: ModalProps) => JSX.Element = ({
  visible,
  setVisible,
  children,
  closeButton,
  paginationProps
}) => {
  const handleModalClose = () => {
    setVisible(false)
    if (paginationProps) {
      paginationProps.setPageNumber(0)
    }
  }

  // TODO - append Modal to ReactDOM.portal
  // TODO - hide Modal on background click
  return (
    <ModalContainer visible={visible}>
      <div className={`${styles.modal} ${visible ? styles.visible : ""}`}>
        <div className={styles.modalContent}>
          {children}
        </div>
        {paginationProps && paginationProps.pageNumber > 0 && <button
          className={`${styles.modalButton} ${styles.backButton}`}
          onClick={() => paginationProps.setPageNumber(0)}
          type="button"
        >
          <BackArrow />
        </button>}
        {closeButton && <button
          className={`${styles.modalButton} ${styles.closeButton}`}
          onClick={handleModalClose}
          type="button"
        >
          <CloseModalIcon />
        </button>}
      </div>
    </ModalContainer>
  );
};
