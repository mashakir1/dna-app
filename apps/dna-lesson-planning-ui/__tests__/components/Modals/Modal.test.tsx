import { screen, fireEvent, render } from "@testing-library/react";

import { Modal, ModalProps } from "components/Modals";

describe("Modal", () => {
  it("renders an empty modal", () => {
    const emptyContent = <></>;
    const modal = renderModal({
      visible: true,
      children: emptyContent
    });

    expect(modal).toMatchSnapshot();
  });

  it("can hide a modal", () => {
    const emptyContent = <></>;
    const modal = renderModal({
      visible: false,
      children: emptyContent
    });

    expect(modal).toMatchSnapshot();
  });

  it("renders content within the modal", () => {
    const content = <span>test</span>;
    const modal = renderModal({
      visible: true,
      children: content
    });

    expect(modal).toMatchSnapshot();
  });

  it("renders the close button", () => {
    const emptyContent = <></>;
    const modal = renderModal({
      visible: true,
      closeButton: true,
      children: emptyContent
    });

    expect(modal).toMatchSnapshot();
  });

  it("handles the close button correctly", () => {
    const emptyContent = <></>;
    const testSetVisible = jest.fn();

    const modal = renderModal({
      visible: true,
      closeButton: true,
      setVisible: testSetVisible,
      children: emptyContent
    });

    expect(modal).toMatchSnapshot();

    fireEvent.click(screen.queryByRole("button"));
    expect(testSetVisible).toHaveBeenCalledWith(false);
  });
});

describe("Paginated Modal", () => {
  test("first page has no buttons", () => {
    const firstPage = <span>firstPage</span>;
    const pagination = {
      pageNumber: 0,
      setPageNumber: () => null
    };

    const modal = renderModal({
      paginationProps: pagination,
      visible: true,
      children: firstPage
    });

    expect(screen.queryByRole("button")).toBeFalsy();
    expect(modal).toMatchSnapshot();
  });

  test("second page has buttons which are clickable", () => {
    const secondPage = <span>secondPage</span>;
    const testSetPageNumber = jest.fn();
    const pagination = {
      pageNumber: 1,
      setPageNumber: testSetPageNumber
    };

    const modal = renderModal({
      paginationProps: pagination,
      visible: true,
      children: secondPage,
    });

    expect(screen.getByRole("button")).toBeTruthy();
    expect(modal).toMatchSnapshot();

    fireEvent.click(screen.queryByRole("button"));
    expect(testSetPageNumber).toHaveBeenCalledWith(0);
  });
});

const renderModal = (props: ModalProps) => {
  return render(<Modal {...props}>{props.children}</Modal>).container;
};
