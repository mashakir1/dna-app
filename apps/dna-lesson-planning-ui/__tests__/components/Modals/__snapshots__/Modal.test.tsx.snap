// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal can hide a modal 1`] = `
<div>
  <div
    class="modalContainer "
  >
    <div
      class="modal "
    >
      <div
        class="modalContent"
      />
    </div>
  </div>
</div>
`;

exports[`Modal handles the close button correctly 1`] = `
<div>
  <div
    class="modalContainer visible"
  >
    <div
      class="modal visible"
    >
      <div
        class="modalContent"
      />
      <button
        class="modalButton closeButton"
        type="button"
      >
        <svg
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M5.31542 5.31542C5.73599 4.89486 6.41786 4.89486 6.83842 5.31542L12 10.477L17.1616 5.31542C17.5821 4.89486 18.264 4.89486 18.6846 5.31542C19.1051 5.73599 19.1051 6.41786 18.6846 6.83842L13.523 12L18.6846 17.1616C19.1051 17.5821 19.1051 18.264 18.6846 18.6846C18.264 19.1051 17.5821 19.1051 17.1616 18.6846L12 13.523L6.83842 18.6846C6.41786 19.1051 5.73599 19.1051 5.31542 18.6846C4.89486 18.264 4.89486 17.5821 5.31542 17.1616L10.477 12L5.31542 6.83842C4.89486 6.41786 4.89486 5.73599 5.31542 5.31542Z"
            fill="#30CCD6"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;

exports[`Modal renders an empty modal 1`] = `
<div>
  <div
    class="modalContainer visible"
  >
    <div
      class="modal visible"
    >
      <div
        class="modalContent"
      />
    </div>
  </div>
</div>
`;

exports[`Modal renders content within the modal 1`] = `
<div>
  <div
    class="modalContainer visible"
  >
    <div
      class="modal visible"
    >
      <div
        class="modalContent"
      >
        <span>
          test
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal renders the close button 1`] = `
<div>
  <div
    class="modalContainer visible"
  >
    <div
      class="modal visible"
    >
      <div
        class="modalContent"
      />
      <button
        class="modalButton closeButton"
        type="button"
      >
        <svg
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M5.31542 5.31542C5.73599 4.89486 6.41786 4.89486 6.83842 5.31542L12 10.477L17.1616 5.31542C17.5821 4.89486 18.264 4.89486 18.6846 5.31542C19.1051 5.73599 19.1051 6.41786 18.6846 6.83842L13.523 12L18.6846 17.1616C19.1051 17.5821 19.1051 18.264 18.6846 18.6846C18.264 19.1051 17.5821 19.1051 17.1616 18.6846L12 13.523L6.83842 18.6846C6.41786 19.1051 5.73599 19.1051 5.31542 18.6846C4.89486 18.264 4.89486 17.5821 5.31542 17.1616L10.477 12L5.31542 6.83842C4.89486 6.41786 4.89486 5.73599 5.31542 5.31542Z"
            fill="#30CCD6"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;

exports[`Paginated Modal first page has no buttons 1`] = `
<div>
  <div
    class="modalContainer visible"
  >
    <div
      class="modal visible"
    >
      <div
        class="modalContent"
      >
        <span>
          firstPage
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Paginated Modal second page has buttons which are clickable 1`] = `
<div>
  <div
    class="modalContainer visible"
  >
    <div
      class="modal visible"
    >
      <div
        class="modalContent"
      >
        <span>
          secondPage
        </span>
      </div>
      <button
        class="modalButton backButton"
        type="button"
      >
        <svg
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M11.1847 4L12.5066 5.48721L6.1141 11.1695H22V13.1593H6.11461L12.5066 18.8411L11.1847 20.3283L2 12.1642L11.1847 4Z"
            fill="#30CCD6"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
